internal/cmd/cmd.go

```diff
@@ -61,9 +66,13 @@ type Cmd struct {
 	// ExitState is filled out after Wait() (or Run() or Output()) completes.
 	ExitState *ExitState
 
+	// afterExitFuns are run after the process exits, but before wait returns and
+	// IO is waited on
+	afterExitFuns []func(context.Context) error
```

Comment:
I guess I'm not very clear on why we should have these over just by default correctly closing stdio every time?

Reply:
I didn't know what our consenses was on that, since we may have chosen the different approach for a reason.
Plus, I think closing stdio every time would cause breaking changes in the code

Reply:
what breaking changes do you anticipate?

Reply:
I think they would be in the shim exec (plus shim diag and runhcs) to monitor for situations when the command finishes but the stdin stream remains open, and closing that.
So mostly around exec-ing, i think

Reply:
Talked a bit offline. We've decided that for now we'll remove the `afterExitFuns` in favor of a bool to indicate whether or not to close stdin.